### PR TITLE
Fc 732 hipay customdata

### PR DESCRIPTION
**Pull request by abailleulfc**
Documentation related to : https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1033#note_840216011

### CODE REVIEW

File: source/docs/advanced/payments/hipay.md [hunk 1]
```diff
@@ -183,4 +196,52 @@ To allow loading HiPay related remote resources:
 
 <blockquote class="note">
 Please keep in mind that HiPay payment methods depends on the Customer country and the Store currency. Different contexts could display different payment methods in the checkout.
-</blockquote>
\ No newline at end of file
+</blockquote>
+
+### Advanced: customize data sent to HiPay
+
+The HiPay payment module is extensible. It leverages Front-Commerce's "data transform" pattern to allow developers to customize payment data sent to HiPay.
```
Comment (paul-vd) [inline source/docs/advanced/payments/hipay.md:203]:
```suggestion
The HiPay payment module is extensible. It leverages the "data transform" pattern from Front-Commerce to allow developers to customize payment data sent to HiPay.
```

File: source/docs/advanced/payments/hipay.md [hunk 2]
```diff
@@ -183,4 +196,52 @@ To allow loading HiPay related remote resources:
...
+  contextEnhancer: ({ loaders }) => {
+    this.loaders.HiPay.registerPaymentDataTransform((paymentData) => ({
+      custom_data: JSON.stringify(paymentData.customData)
+    }))
+  },
+};
+```
+
+<blockquote class="warning">
+
```
Comment (paul-vd) [inline source/docs/advanced/payments/hipay.md:244]: **non-blocking :** remove additional line

```suggestion
```

File: source/docs/advanced/payments/hipay.md [hunk 3]
```diff
@@ -183,4 +196,52 @@ To allow loading HiPay related remote resources:
...
+**Warning:** root attributes like `custom_data` indicated here are directly added to the HiPay `/order` form-data ([see the webservice documentation](https://developer.hipay.com/api-explorer/api-online-payments#/payments/requestNewOrder)). Add other attributes cautiously.
+
```
Comment (paul-vd) [inline source/docs/advanced/payments/hipay.md:246]: **non-blocking :** remove additional line

```suggestion
```

**Review by paul-vd — APPROVED** [review]

File: source/docs/advanced/payments/hipay.md [hunk 4]
```diff
@@ -183,4 +196,30 @@ To allow loading HiPay related remote resources:
...
+The HiPay payment module is extensible. It leverages Front-Commerce's "data transform" pattern to allow developers to customize payment data sent to HiPay.
+
+You need to create a backend loader to register the transformation.
+
+```
+export default {
+  namespace: "HiPay/Custom",
+  dependencies: ["HiPay/Core"],
+  contextEnhancer: ({ loaders }) => {
+    this.loaders.HiPay.registerPaymentDataTransform((customer, order) => ({
```
Comment (real34) [inline source/docs/advanced/payments/hipay.md:212]:
```suggestion
    loaders.HiPay.registerPaymentDataTransform((customer, order) => ({
```

**Review by real34 — APPROVED** [review]

**Pull request merged**